rspace/src/main/scala/coop/rchain/rspace/channelStore/instances/ChannelStoreImpl.scala

```diff
@@ -17,32 +17,38 @@ import scodec.codecs.{discriminated, uint2}
 object ChannelStoreImpl {
   def apply[F[_]: Sync, C](
       store: KeyValueStore[F],
-      sc: Serialize[C],
-      codecC: Codec[C]
+      sc: Serialize[C]
   ) =
-    ChannelStoreImpl(store.toTypedStore(codecPureBlake2b256Hash, codecChannelHash), sc, codecC)
+    ChannelStoreImpl(store.toTypedStore(codecPureBlake2b256Hash, codecChannelHash), sc)
 
   final case class ChannelStoreImpl[F[_]: Sync, C](
       store: KeyValueTypedStore[F, Blake2b256Hash, ChannelHash],
-      sc: Serialize[C],
-      codecC: Codec[C]
+      sc: Serialize[C]
   ) extends ChannelStore[F, C] {
-    implicit val serializeC: Serialize[C]    = sc
-    val continuationSerializeC: Serialize[C] = Serialize.fromCodec(codecC)
-
     override def putChannelHash(channel: C): F[Unit] =
       for {
-        eventKey <- StableHashProvider.hash(channel)(sc).pure[F]
-        dataHash = hashDataChannel(channel, codecC)
-        joinHash = hashJoinsChannel(channel, codecC)
+        // C => hash(C)
+        eventKey <- Sync[F].delay(StableHashProvider.hash(channel)(sc))
+
+        // C => hashPrefix(C)
+        dataHash = hashDataChannel(channel, sc)
+        joinHash = hashJoinsChannel(channel, sc)
         _        <- store.put(eventKey, DataJoinHash(dataHash, joinHash))
       } yield ()
 
     override def putContinuationHash(channels: Seq[C]): F[Unit] =
       for {
-        channelsHashes   <- toOrderedByteVectors(channels).map(Blake2b256Hash.create).pure[F]
-        eventKey         = continuationKey(channelsHashes)
-        continuationHash = hashContinuationsChannels(channels, continuationSerializeC)
+        // Hash each channel in a list
+        channelsHashes <- Sync[F].delay(
```

Comment:
I am just asking a question because you told me before. `delay` is different from `pure`, right? Mostly we want to suspend our function as much as possible which would make the function as lazy as possible, right?

Reply:
The main difference is that `delay` argument is defined as a thunk which is like anonymous function.
When calling delay `A` don't needs to be evaluated until `f` is requested which is done by monix when executing `F`.
 `delay(f: => A) = F[A]`
But calling pure will evaluate `A` immediately and wrap in `F`
`pure(a: A) = F[A]`